app/src/main/java/com/omkarmoghe/pokemap/views/map/MapWrapperFragment.java

```diff
@@ -164,24 +169,45 @@ private void setPokemonMarkers(final List<CatchablePokemon> pokeList){
                 markerList = new ArrayList<Marker>(); //cleaning the array
             }
 
-            for (CatchablePokemon poke : pokeList) {
-                int resourceID = getResources().getIdentifier("p" + poke.getPokemonId().getNumber(), "drawable", getActivity().getPackageName());
-                long millisLeft = poke.getExpirationTimestampMs() - System.currentTimeMillis();
-                Marker marker = mGoogleMap.addMarker(new MarkerOptions()
-                        .position(new LatLng(poke.getLatitude(), poke.getLongitude()))
-                        .title(poke.getPokemonId().name())
-                        .snippet("Dissapears in: " + getDurationBreakdown(millisLeft))
-                        .icon(BitmapDescriptorFactory.fromResource(resourceID)));
-
-                marker.showInfoWindow();
-                //adding pokemons to list to be removed on next search
-                markerList.add(marker);
+            for (final CatchablePokemon poke : pokeList) {
+                //Showing images using glide
+                Glide.with(getActivity())
+                        .load("http://serebii.net/pokemongo/pokemon/"+getCorrectPokemonIdNumber(poke.getPokemonId().getNumber())+".png")
+                        .asBitmap()
+                        .skipMemoryCache(false)
+                        .diskCacheStrategy(DiskCacheStrategy.ALL)
+                        .into(new SimpleTarget<Bitmap>(120, 120) { // Width and height FIXME: Maybe get different sizes based on devices DPI?
+                            @Override
+                            public void onResourceReady(Bitmap bitmap, GlideAnimation anim) {
+                                //Setting marker since we got image
+                                long millisLeft = poke.getExpirationTimestampMs() - System.currentTimeMillis();
+                                Marker marker = mGoogleMap.addMarker(new MarkerOptions()
+                                        .position(new LatLng(poke.getLatitude(), poke.getLongitude()))
+                                        .title(poke.getPokemonId().name())
+                                        .snippet("Dissapears in: " + getDurationBreakdown(millisLeft))
+                                        .icon(BitmapDescriptorFactory.fromBitmap(bitmap)));
+                                //Not showing infowindow anymore, let use choose wich to open
+                                markerList.add(marker);
+                            }
+                        });
             }
         } else {
             Toast.makeText(getContext(), "The map is not initialized.", Toast.LENGTH_LONG).show();
         }
     }
 
+    //Getting correct pokemon Id eg: 1 must be 001, 10 must be 010
+    private String getCorrectPokemonIdNumber (int pokemonId){
```

Comment:
I'd replace this unstable manual converter with usage of `String.format("%03d", pokemonId)` for now